cpu/saml21/cpu.c

```diff
@@ -296,8 +297,13 @@ void cpu_init(void)
         }
     }
     /* clock used by timers */
+#if (CLOCK_CORECLOCK == 48000000U)
+#define CLOCK_OSC16M    16000000U
+#else
+#define CLOCK_OSC16M    CLOCK_CORECLOCK
+#endif
```

Comment:
but you have 

```C
    /* set OSC16M according to CLOCK_CORECLOCK */
#if (CLOCK_CORECLOCK == 48000000U) || (CLOCK_CORECLOCK == 16000000U)
    OSCCTRL->OSC16MCTRL.bit.FSEL = OSCCTRL_OSC16MCTRL_FSEL_16_Val;
#elif (CLOCK_CORECLOCK == 12000000U)
    OSCCTRL->OSC16MCTRL.bit.FSEL = OSCCTRL_OSC16MCTRL_FSEL_12_Val;
#elif (CLOCK_CORECLOCK == 8000000U)
    OSCCTRL->OSC16MCTRL.bit.FSEL = OSCCTRL_OSC16MCTRL_FSEL_8_Val;
#elif (CLOCK_CORECLOCK == 4000000U)
    OSCCTRL->OSC16MCTRL.bit.FSEL = OSCCTRL_OSC16MCTRL_FSEL_4_Val;
#else
#error "Please select a valid CPU frequency"
#endif
```

So why not

```C
    /* set OSC16M according to CLOCK_CORECLOCK */
#if (CLOCK_CORECLOCK == 48000000U) || (CLOCK_CORECLOCK == 16000000U)
#define OSC16MCTRL_FSEL OSCCTRL_OSC16MCTRL_FSEL_16_Val
#define OSC16MCTRL_HZ   MHZ(16)
#elif (CLOCK_CORECLOCK == 12000000U)
#define OSC16MCTRL_FSEL OSCCTRL_OSC16MCTRL_FSEL_12_Val
#define OSC16MCTRL_HZ   MHZ(12)
#elif (CLOCK_CORECLOCK == 8000000U)
#define OSC16MCTRL_FSEL OSCCTRL_OSC16MCTRL_FSEL_8_Val
#define OSC16MCTRL_HZ   MHZ(8)
#elif (CLOCK_CORECLOCK == 4000000U)
#define OSC16MCTRL_FSEL OSCCTRL_OSC16MCTRL_FSEL_4_Val
#define OSC16MCTRL_HZ   MHZ(4)
#else
#error "Please select a valid CPU frequency"
#endif

and then have

```C
OSCCTRL->OSC16MCTRL.bit.FSEL = OSC16MCTRL_FSEL;
…
_gclk_setup(SAM0_GCLK_TIMER, GCLK_GENCTRL_GENEN | GCLK_GENCTRL_SRC_OSC16M
            | GCLK_GENCTRL_DIV(OSC16MCTRL_HZ/sam0_gclk_freq(SAM0_GCLK_TIMER)));
```